_posts/en/newsletters/2019-05-14-newsletter.md

```diff
@@ -0,0 +1,597 @@
+right now: single-sig P2PKH and P2WPKH spends, which constitute FIXME:x%
+of all spends.  FIXME:p2sh.info-ref
+
+Single-sig P2WPKH wallets today currently generate a private key, derive
+a public key (pubkey) from it, and hash that pubkey to create the
+witness program for a bech32 address.  (P2PKH is functionally identical
+but uses a different script and a different address encoding.)
+
+| Object      | Operation                                                       | Example result |
+|-|-|-|
+| Private key | read 32 bytes from [CSPRNG][], or using [BIP32][] HD derivation | 0xFIXME |
+| Public key  | point(0xFIXME), or using [BIP32][] HD public derivation         | 0x02FIXME |
+| Hash        | ripemd(sha256(0x02FIXME))                                       | 0xFIXME |
+| Address     | bech32.encode(0, 0xFIXME)                                       | bc1qFIXME |
+
+Under Taproot, the hashing step will be skipped and so the bech32
+address will contain the pubkey directly, with one small change.
+Currently, 33-byte Bitcoin-style pubkeys are encoded to start with either
+a 0x02 or 0x03 to allow validators to reconstruct the key's Y-coordinate
+on the secp256k1 elliptic curve; in bip-taproot, the value of this byte
+is reduced by two so that 0x02 becomes 0x00 and 0x03 becomes 0x01,
+although the meaning is exactly the same.  Also the witness version is
+changed from the `0` used for P2WPKH/P2WSH to a `1`.
+
+| Object           | Operation                                                       | Example result |
+| Private key      | read 32 bytes from [CSPRNG][], or using [BIP32][] HD derivation | 0xFIXME |
+| Public key       | point(0xFIXME), or using [BIP32][] HD public derivation         | 0x02FIXME |
+| Alter key prefix | (key[0] - 2),key[1:33])                                         | 0x00FIXME |
+| Address          | bech32.encode(1, 0x00FIXME)                                     | bc1pFIXME |
+
+Here's an example of existing addresses compared to an example taproot
+address.
+
+    p2pkh FIXME
+    p2sh FIXME
+    p2wpkh FIXME
+    p2wsh FIXME
+    taproot FIXME
+
+Spending from P2PKH or P2WPKH requires the spender include their public
+key in their input.  For Taproot, the public key was provided in the
+UTXO being spent, so several vbytes are saved by not including it again.
+The spend must also include a signature; this is a Schnorr-style
+signature as defined by [bip-schnorr][] with an optional sighash byte
+appended.  If the default sighash is used, the signature is 64 bytes (16
+vbytes); if a non-default is used, it is 65 bytes (16.25
+vbytes[^vbytes-decimal]).  Overall, this makes the cost to create and
+spend a Taproot single-sig output about 1% more expensive than P2WPKH.
+However, the costs are distributed differently---it costs more to create
+a Taproot output than a P2WPKH output and less to spend it, which may
+help contribute towards keeping the size of the UTXO set manageable.
```

Comment:
It's worth comparing with native P2WSH where people are already fine with sending to a 32-byte witness program (and I don't think I've ever heard someone complain about that). Also, one idea for a change to the BIP (something @jnewbery brought up) is dropping the 0/1 prefix byte and just putting the X coordinate in the output. Obviously not something to put in the description right now, but it may help color things.